### PR TITLE
Use array as ResourceGroupId's JSON representation

### PR DESCRIPTION
**Pull request by rongrong**

### CODE REVIEW

**Comment by electrum** [comment]
Instead of making the commit message just be "Revert XXX", make it a normal commit with an explanation of the change (including the why).

**Comment by electrum** [comment]
Nit: capitalize "JSON" in commit messages

File: presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/ResourceGroupId.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public ResourceGroupId(ResourceGroupId parent, String name)
         return result;
     }
 
-    private ResourceGroupId(List<String> segments)
+    public ResourceGroupId(@JsonProperty("segments") List<String> segments)
```
Comment (electrum) [inline presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/ResourceGroupId.java]: Let's make the JSON representation simply be an array, not an object with a single array field.

**Comment by rongrong** [comment]
I'm inclined to change the ResourceGroupId.toString() to be something more obviously describe the hierarchy as well. Any suggestions?

File: presto-spi/src/test/java/com/facebook/presto/spi/resourceGroups/TestResourceGroupId.java [hunk 2]
```diff
@@ -28,6 +30,14 @@ public void testBasic()
     }
 
     @Test
+    public void testCodec()
```
Comment (electrum) [inline presto-spi/src/test/java/com/facebook/presto/spi/resourceGroups/TestResourceGroupId.java]: Add a test that compares to a fixed JSON string (so we can see the exact representation and make sure it doesn't accidentally change).

**Comment by electrum** [comment]
Wrap text descriptions to ~70 characters. (longer lines are fine if they are something like a benchmark result table)

**Comment by electrum** [comment]
No need to mention the "this reverts commit"

File: presto-spi/src/test/java/com/facebook/presto/spi/resourceGroups/TestResourceGroupId.java [hunk 3]
```diff
@@ -27,14 +23,8 @@
     @Test
     public void testBasic()
     {
-        new ResourceGroupId("test_test");
```
Comment (electrum) [inline presto-spi/src/test/java/com/facebook/presto/spi/resourceGroups/TestResourceGroupId.java]: Leave this test

**Pull request merged**